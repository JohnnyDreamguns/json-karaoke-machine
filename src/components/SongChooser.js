import React from 'react';

const SongChooser = ({
  play, tempo, handleChangeTempo, isPlaying
}) => (
  <header className="masthead">
    <div className="container h-100">
      <div className="row h-100">
        <div className="col-lg-7 my-auto">
          <div className="header-content mx-auto">
            <h1 className="mb-5">Press play to hear the sweet 
              sounds of the Web Audio API</h1>

            <div className="header-controls">
              <button className="btn btn-outline btn-xl" data-playing="false" 
                role="switch" aria-checked="false" onClick={play}>
                { isPlaying ? 'Stop' : 'Play!' }
              </button>

              <div className="tempo-control">
                <label forhtml="tempo">{tempo}bpm</label>
                <input type="range" id="tempo" min="70" max="250" value={tempo} step="1" 
                  onChange={handleChangeTempo} />
              </div>
            </div>
          </div>
        </div>
        <div className="col-lg-5 my-auto">
          <h3>Now Playing: Jump by Van Halen</h3>
          <img src={require('/img/vanhalen.jpg')} className="img-fluid" alt="" />
        </div>
      </div>
    </div>
  </header>
);

export default SongChooser;