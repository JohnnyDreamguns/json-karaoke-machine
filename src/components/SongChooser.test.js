import 'babel-polyfill';
import React from 'react';
import SongChooser from './SongChooser';
import renderer from 'react-test-renderer';

const defaultProps = {
  play: jest.fn(),
  isPlaying: false,
  tempo: 120,
  handleChangeTempo: jest.fn()
};

describe('SongChooser', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SongChooser {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
