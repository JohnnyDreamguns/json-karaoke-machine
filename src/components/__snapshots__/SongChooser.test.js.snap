// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongChooser renders correctly 1`] = `
<header
  className="masthead"
>
  <div
    className="container h-100"
  >
    <div
      className="row h-100"
    >
      <div
        className="col-lg-7 my-auto"
      >
        <div
          className="header-content mx-auto"
        >
          <h1
            className="mb-5"
          >
            Press play to hear the sweet sounds of the Web Audio API
          </h1>
          <div
            className="header-controls"
          >
            <button
              aria-checked="false"
              className="btn btn-outline btn-xl"
              data-playing="false"
              onClick={[MockFunction]}
              role="switch"
            >
              Play!
            </button>
            <div
              className="tempo-control"
            >
              <label
                forhtml="tempo"
              >
                120
                bpm
              </label>
              <input
                id="tempo"
                max="250"
                min="70"
                onChange={[MockFunction]}
                step="1"
                type="range"
                value={120}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-5 my-auto"
      >
        <h3>
          Now Playing: Jump by Van Halen
        </h3>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </div>
</header>
`;
